src/components/PicklistOption/__test__/picklistOption.spec.js

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import { mount, shallow } from 'enzyme';
+import { Option as PicklistOption } from '../index';
+
+describe('<PicklistOption />', () => {
+    it('should have the right classnames when active', () => {
+        const component = mount(<PicklistOption label="option 1" name="option1" />);
+        expect(component.find('li').prop('className')).toBe('rainbow-picklist-option');
+        component.setProps({
+            activeOptionName: 'option1',
+        });
+        expect(component.find('li').prop('className')).toBe(
+            'rainbow-picklist-option rainbow-picklist-option_active',
+        );
+    });
+    it('should have the right classnames when selected', () => {
+        const component = mount(<PicklistOption label="option 1" name="option1" />);
+        expect(component.find('li').prop('className')).toBe('rainbow-picklist-option');
+        component.setProps({
+            currentValueName: 'option1',
+        });
+        expect(component.find('li').prop('className')).toBe(
+            'rainbow-picklist-option rainbow-picklist-option_selected',
+        );
+    });
+    it('should render a option header when variant is header', () => {
+        const component = mount(<PicklistOption label="Header 1" variant="header" />);
+        expect(component.find('li').prop('className')).toBe('rainbow-picklist-option_header');
+        expect(component.find('li > span.rainbow-picklist-option_header-label').exists()).toBe(
+            true,
+        );
+    });
+    it('should render a regular option when variant is default', () => {
+        const component = mount(<PicklistOption label="option 1" name="option1" />);
+        expect(component.find('li').prop('className')).toBe('rainbow-picklist-option');
+    });
+    it('should not register when is disabled', () => {
+        const optionRegisterFn = jest.spyOn(PicklistOption.prototype, 'register');
+        mount(<PicklistOption disabled />);
+        expect(optionRegisterFn).not.toHaveBeenCalled();
+        optionRegisterFn.mockClear();
+    });
+    it('should not register when is variant header', () => {
+        const optionRegisterFn = jest.spyOn(PicklistOption.prototype, 'register');
```

Comment:
if this is like a global mock function then try to create it only once and then reset inside a beforeEach hook in order to not repeat the same too much